src/source_code_style.rs

```diff
@@ -171,17 +171,17 @@ fn detect_quote(contents: &str, locator: &SourceCodeLocator) -> Option<Quote> {
 
 /// Detect the line ending style of the given contents.
 fn detect_line_ending(contents: &str) -> Option<LineEnding> {
-    if let Some(position) = contents.find('\n') {
-        let position = position.saturating_sub(1);
-        return if let Some('\r') = contents.chars().nth(position) {
+    return if let Some(position) = contents.find('\r') {
+        if let Some('\n') = contents.chars().nth(position + 1) {
             Some(LineEnding::CrLf)
         } else {
-            Some(LineEnding::Lf)
-        };
-    } else if contents.find('\r').is_some() {
-        return Some(LineEnding::Cr);
-    }
-    None
+            Some(LineEnding::Cr)
```

Comment:
Oh, in Cargo.toml, we should also disable the `cr_lines` feature on Ropey, I think? Or, actually, we could keep it on and consider using `Ropey` to find the first line ending quickly?

Reply:
I completely missed that `ropey` seems to have proper support for `CR` line endings. Perhaps we can support them after all.. I'll check that.

Reply:
But I think things like `.lines()` still won't work, right? Per your original note in the issue.